torch/_inductor/fx_passes/decompose_mem_bound_mm.py

```diff
@@ -57,7 +59,13 @@ def should_decompose_bmm(mat1, mat2) -> bool:
             mat1.shape[2] < max_other_dimention_decomposition
         ) + (mat2.shape[2] < max_other_dimention_decomposition) < 2:
             return False
-    return True
+        return True
+    if check_device(mat1, mat2, device="cpu"):
+        if len(mat1.shape) != 3 or len(mat2.shape) != 3:
+            return False
+        if mat1.shape[0] == 1 and mat2.shape[0] == 1:
+            return True
```

Comment:
I know it's already being done this way in the pass but nonetheless could we use guard_shape_oblivious for the new checks youve added ?